src/main/resources/templates/db_scripts/schema.json

```diff
@@ -2,7 +2,7 @@
+      "snippet": "CREATE OR REPLACE VIEW receiving_history_view AS SELECT pieces.id AS id, json_build_object('id', pieces.jsonb->>'id', 'caption', pieces.jsonb->>'caption', 'comment', pieces.jsonb->>'comment', 'dateOrdered', purchase_order.jsonb->>'dateOrdered', 'checkin', po_line.jsonb->>'checkinItems', 'instanceId', po_line.jsonb->>'instanceId','itemId', pieces.jsonb->>'itemId', 'locationId', pieces.jsonb->>'locationId', 'pieceFormat', pieces.jsonb->>'format', 'poLineId', pieces.jsonb->>'poLineId', 'poLineNumber', po_line.jsonb->>'poLineNumber', 'purchaseOrderId', po_line.jsonb->>'purchaseOrderId', 'poLineReceiptStatus', po_line.jsonb->>'receiptStatus','receivedDate', pieces.jsonb->>'receivedDate', 'title', po_line.jsonb->>'title','receivingNote', po_line.jsonb->'details'->>'receivingNote', 'receivingStatus', pieces.jsonb->>'receivingStatus', 'supplement', pieces.jsonb->>'supplement') AS jsonb, COALESCE(acquisitions_unit_assignments.jsonb, '{}'::jsonb) || json_build_object('id', pieces.jsonb->>'id', 'caption', pieces.jsonb->>'caption', 'comment', pieces.jsonb->>'comment', 'dateOrdered', purchase_order.jsonb->>'dateOrdered', 'checkin', po_line.jsonb->>'checkinItems', 'instanceId', po_line.jsonb->>'instanceId','itemId', pieces.jsonb->>'itemId', 'locationId', pieces.jsonb->>'locationId', 'pieceFormat', pieces.jsonb->>'format', 'poLineId', pieces.jsonb->>'poLineId', 'poLineNumber', po_line.jsonb->>'poLineNumber', 'purchaseOrderId', po_line.jsonb->>'purchaseOrderId', 'poLineReceiptStatus', po_line.jsonb->>'receiptStatus','receivedDate', pieces.jsonb->>'receivedDate', 'title', po_line.jsonb->>'title','receivingNote', po_line.jsonb->'details'->>'receivingNote', 'receivingStatus', pieces.jsonb->>'receivingStatus', 'supplement', pieces.jsonb->>'supplement')::jsonb AS metadata FROM pieces LEFT OUTER JOIN po_line ON pieces.jsonb->>'poLineId' = po_line.jsonb->>'id' LEFT OUTER JOIN purchase_order ON po_line.jsonb->>'purchaseOrderId' = purchase_order.jsonb->>'id' LEFT OUTER JOIN acquisitions_unit_assignments ON acquisitions_unit_assignments.jsonb->>'recordId' = purchase_order.jsonb->>'id';",
```

Comment:
I think for simplicity you can use following to build metadata
```sql
COALESCE(acquisitions_unit_assignments.jsonb, '{}'::jsonb) || po_line.jsonb || purchase_order.jsonb || pieces.jsonb AS metadata
```

This will allow to search by additional properties but I think it should be okay...

Reply:
At json_build_object() function several fields were renamed for example:
```
receiptStatus->poLineReceiptStatus
format-> pieceFormat
```
That mean if we will concatenate the tables we will can't search by poLineReceiptStatus or pieceFormat. And all attempts will return empty list.